source/slang/slang-ir-operator-shift-overflow.cpp

```diff
@@ -0,0 +1,66 @@
+// slang-ir-operator-shift-overflow.cpp
+#include "slang-ir-operator-shift-overflow.h"
```

Comment:
you might want to rename this into something more generic, so that additional checks can be added here. 

lets say, if you want to check for valid integer promotions for various other operators, other than shift..

Reply:
I may need to do it to handle the other operators.
But the approach for the other operators are not decided yet so I wouldn't be preemptive about it yet.